sample-app/src/components/InputDropdown.tsx

```diff
@@ -97,24 +131,43 @@ export default function InputDropdown({
       evt.preventDefault();
     }
 
+    const isFirstSectionFocused = focusedSectionIndex === 0;
+    const isLastSectionFocused = focusedSectionIndex === options.length - 1;
     const isFirstOptionFocused = focusedOptionIndex === 0;
-    const isLastOptionFocused = focusedOptionIndex === options.length - 1;
+    const isLastOptionFocused = 
+      focusedSectionIndex !== undefined &&
+      focusedOptionIndex === options[focusedSectionIndex].results.length - 1;
     if (evt.key === 'Enter') {
-      updateInputValue(inputValue);
-      onSubmit(inputValue);
-      dispatch({ type: 'HideOptions' });
+      if (!onlySubmitOnOption || focusedOptionIndex !== undefined) {
```

Comment:
what is onlySubmitOnOption, and why do we need it?

Reply:
when onlySubmitOnOption is true, the input should not be submitted when 'enter' is pressed, unless it is focused on an option. otherwise, if onlySubmitOnOption is false, when 'enter' is pressed, the user input is submitted if no option is focused.

if we dispatch `HideOptions` regardless, then if a user types something into the input bar for filter search and presses 'enter' without focusing on a filter option, the dropdown would be hidden but no filter would be applied. is that the behavior we want, or would we want to keep the dropdown open?

Reply:
I see, can you check what does sdk do? I think it makes sense to close it if user press enter without focusing on an option

Reply:
the SDK leaves it open